src/index/blockfilterindex.h

```diff
@@ -30,6 +33,9 @@ class BlockFilterIndex final : public BaseIndex
     bool ReadFilterFromDisk(const FlatFilePos& pos, BlockFilter& filter) const;
     size_t WriteFilterToDisk(FlatFilePos& pos, const BlockFilter& filter);
 
+    Mutex m_cs_headers_cache;
+    std::map<uint256, uint256> m_headers_cache GUARDED_BY(m_cs_headers_cache);
```

Comment:
Any reason `std::unordered_map` can't be used? I see that the tree-based `std::map` and `std::set` are used primarily throughout the codebase, but figured it may just be historical. Is there another reason these are preferred?

Reply:
I expect mostly laziness. With an `unordered_map`, you need to provide a hash function. In this case, it's easy. The data is all hash digests from block chain data, so I can just take the first 8 bytes from the uint256 filter header.

Reply:
Just as a reminder that there is no free lunch and unordered_map comes with a size penalty

Reply:
I've reimplemented this as an unordered_map